### PR TITLE
Added currency field to course run search api

### PR DESCRIPTION
**Pull request by manan-memon**
This PR adds the currency code field for `first_enrollable_paid_seat_price` which is fetched from e-commerce for a paid verified course.

JIRA: https://edlyio.atlassian.net/browse/EDLY-6981

### CODE REVIEW

File: course_discovery/apps/course_metadata/models.py [hunk 1]
```diff
@@ -1414,6 +1414,17 @@ def first_enrollable_paid_seat_price(self):
 
         price = int(seats[0].price) if seats[0].price else None
         return price
+    
+    @property
+    def first_enrollable_paid_seat_price_currency(self):
```
Comment (muhammadali286) [inline course_discovery/apps/course_metadata/models.py]: `first_enrollable_paid_seat_price_currency` -> `first_enrollable_paid_seat_currency`

File: course_discovery/apps/course_metadata/models.py [hunk 2]
```diff
@@ -1414,6 +1414,17 @@ def first_enrollable_paid_seat_price(self):
...
+        # Sort in python to avoid an additional request to the database for order_by
```
Comment (Anas-hameed) [inline course_discovery/apps/course_metadata/models.py]: nit: you can replace # comment with docstrings.

File: course_discovery/apps/course_metadata/models.py [hunk 3]
```diff
@@ -1414,6 +1414,17 @@ def first_enrollable_paid_seat_price(self):
...
+        seats = sorted(self._enrollable_paid_seats(), key=self._upgrade_deadline_sort)
+        if not seats:
+            # Enrollable paid seats are not available for this CourseRun.
```
Comment (Anas-hameed) [inline course_discovery/apps/course_metadata/models.py]: nit: We can remove the inline comment

File: course_discovery/apps/course_metadata/models.py [hunk 4]
```diff
@@ -1414,6 +1414,17 @@ def first_enrollable_paid_seat_price(self):
...
+            return None
+
+        currency = seats[0].currency.code if seats[0].currency.code else None
+        return currency
```
Comment (muhammadali286) [inline course_discovery/apps/course_metadata/models.py]: Lets directly return the value `return seats[0].currency.code or None`

**Review by muhammadali286 — CHANGES_REQUESTED** [review]

File: course_discovery/apps/course_metadata/models.py (hunk 2 above)
Comment (manan-memon) [reply course_discovery/apps/course_metadata/models.py]: Replaced # with docstrings

File: course_discovery/apps/course_metadata/models.py (hunk 3 above)
Comment (manan-memon) [reply course_discovery/apps/course_metadata/models.py]: Removed the inline comment

File: course_discovery/apps/course_metadata/models.py (hunk 1 above)
Comment (manan-memon) [reply course_discovery/apps/course_metadata/models.py]: Changed the field name to `first_enrollable_paid_seat_currency` in all instances.

File: course_discovery/apps/course_metadata/models.py (hunk 4 above)
Comment (manan-memon) [reply course_discovery/apps/course_metadata/models.py]: Updated to return the value directly

**Review by muhammadali286 — APPROVED** [review]

**Review by Anas-hameed — APPROVED** [review]
LGTM

**Review by Anas-hameed — APPROVED** [review]
LGTM

**Comment by manan-memon** [comment]
Closing this PR as there is no need for this, required changes already merged in https://github.com/edly-io/course-discovery/pull/92

**Pull request closed without merging**